src/SFML/Audio/SoundFileReaderOpus.cpp

```diff
@@ -0,0 +1,194 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2016 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Audio/SoundFileReaderOpus.hpp>
+#include <SFML/System/MemoryInputStream.hpp>
+#include <SFML/System/Err.hpp>
+#include <algorithm>
+#include <cctype>
+#include <cassert>
+#include <climits>
+
+namespace
+{
+    int read(void* data, unsigned char* ptr, int bytes)
+    {
+        sf::InputStream* stream = static_cast<sf::InputStream*>(data);
+        return static_cast<int>(stream->read(ptr, bytes));
+    }
+
+    int seek(void* data, opus_int64 offset, int whence)
+    {
+        sf::InputStream* stream = static_cast<sf::InputStream*>(data);
+        switch (whence)
+        {
+            case SEEK_SET:
+                break;
+            case SEEK_CUR:
+                offset += stream->tell();
+                break;
+            case SEEK_END:
+                offset = stream->getSize() - offset;
+        }
+
+        // Return value expected from libopusfile: 0 - Success and -1 - Failure
+        return static_cast<int>(stream->seek(offset)) >= 0 ? 0 : -1;
+    }
+
+    opus_int64 tell(void* data)
+    {
+        sf::InputStream* stream = static_cast<sf::InputStream*>(data);
+        return static_cast<opus_int64>(stream->tell());
+    }
+
+    static OpusFileCallbacks callbacks = {&read, &seek, &tell, NULL};
```

Comment:
`static` is not needed; `callbacks` is defined in an anonymous namespace and thus already has internal linkage.

I'd consider making it `const` however.